File: Codes/Hash/Non-Cryptographic/SDBMHash/SDBMHash.pas
{
    Archive of Reversing.ID
    Non-Cryptographic Hash
}
function SDBMHash(const key : String) : Cardinal;
var
    i : Cardinal;
begin
    Result := 0;
    
    for i := 1 to Length(key) do
    begin
        Result := Ord(key[i]) + (Result shl 6) + (Result shl 16) - Result;
    end;
end;

File: Codes/Hash/Non-Cryptographic/SDBMHash/SDBMHash.vb
'  Archive of Reversing.ID
'  Non-Cryptographic Hash
'

' Fungsi ini merupakan method dari sebuah class
Public Shared Function APHash (key As String) As UInteger
    Dim hash As ULong = 0
	Dim i As UInteger = 0

	For i = 0 To str.Length - 1
		hash = ((CByte(AscW(str(CInt(i)))) + (hash << 6) + (hash << 16) - hash) And UInteger.MaxValue)
	Next

	Return hash
End Function